### PR TITLE
Reactor overhaul part 3

### PR DESCRIPTION
**Pull request by Erik3003**
ONLY MERGE when the new durability system of part 2 has been tested, as this PR implements the new system for IC2 Coolant, because that is needed for neutron moderators.

Added Neutron Moderators, allowing for a lot of fuels to be used in supercritical reactors.
Added a lot of new coolants.
Added new byproducts from depleted fuel rods.
Fixed a bug where Geiger counter would not work on 1x1 reactors.
Made reactors automateable.
Fixed the formatting errors.

### CODE REVIEW

**Comment by GregoriusT** [comment]
You still did not turn off the Autoformatting in its entirety, if Files are changed more than they need to be then that is a sure sign that there is something changing the formatting.

**Comment by Erik3003** [comment]
It's only the empty lines that get changed, not really anything I can do about it.

**Comment by GregoriusT** [comment]
I am 100% sure it did not do that in previous commits before, so what did you change on your Setup to make that happen?

**Comment by OvermindDL1** [comment]
It looks like it's just pruning end-of-line whitespace, this is extremely common as end-of-line whitespace breaks not only expected cursor positions but also some posix'y tools, so it's enforced disabled in almost every text editor out, and IDE's are just fancy text editors so it looks right?

**Comment by OvermindDL1** [comment]
However if it bugs you to look at then set the PR view to ignore whitespace change by adding `&w=1` to the url query or selecting `Hide Whitespace Changes` in the tools (gear) menu, you'll get this:
https://github.com/GregTech6/gregtech6/pull/7/files?diff=split&w=1

Oh, also I use split diff view, for the usual 'unified' view that would instead be:
https://github.com/GregTech6/gregtech6/pull/7/files?diff=unified&w=1

**Comment by GregoriusT** [comment]
Wouldn't it be better for the PR itself to not contain whitespace changes on githubs end, instead of only hiding them from Greg? Also I explained to you plenty of times how broken stuff is when such whitespaces aren't there because Eclipse.

**Comment by OvermindDL1** [comment]
Eclipse issues are not related to file issues.  ^.^

File: src/main/java/gregapi/block/multitileentity/MultiTileEntityItemInternal.java [hunk 1]
```diff
@@ -37,14 +37,7 @@
...
 import gregapi.data.TD;
-import gregapi.item.CreativeTab;
-import gregapi.item.IItemColorableRGB;
-import gregapi.item.IItemEnergy;
-import gregapi.item.IItemGT;
-import gregapi.item.IItemNoGTOverride;
-import gregapi.item.IItemReactorRod;
-import gregapi.item.IItemRottable;
-import gregapi.item.IItemUpdatable;
+import gregapi.item.*;
```
Comment (Speiger) [inline src/main/java/gregapi/block/multitileentity/MultiTileEntityItemInternal.java:40]: Is that a good idea?

File: src/main/java/gregtech/tileentity/sensors/MultiTileEntityGeigerCounter.java [hunk 2]
```diff
@@ -56,10 +57,12 @@ public long getCurrentMax(DelegatorTileEntity<TileEntity> aDelegator) {
 			MultiTileEntityReactorCore TE = (MultiTileEntityReactorCore)aDelegator.mTileEntity;
 			if (FL.Coolant_IC2.is(TE.mTanks[0])) return TE.mTanks[0].getCapacity();
 			int tMaximum = 0;
-			tMaximum += TE.slotHas(0) && ST.item(TE.slot(0)) instanceof IItemReactorRod ? ((IItemReactorRod) ST.item(TE.slot(0))).getReactorRodNeutronMaximum(TE, 0, TE.slot(0)) : 0;
-			tMaximum += TE.slotHas(1) && ST.item(TE.slot(1)) instanceof IItemReactorRod ? ((IItemReactorRod) ST.item(TE.slot(1))).getReactorRodNeutronMaximum(TE, 1, TE.slot(1)) : 0;
-			tMaximum += TE.slotHas(2) && ST.item(TE.slot(2)) instanceof IItemReactorRod ? ((IItemReactorRod) ST.item(TE.slot(2))).getReactorRodNeutronMaximum(TE, 2, TE.slot(2)) : 0;
-			tMaximum += TE.slotHas(3) && ST.item(TE.slot(3)) instanceof IItemReactorRod ? ((IItemReactorRod) ST.item(TE.slot(3))).getReactorRodNeutronMaximum(TE, 3, TE.slot(3)) : 0;
+			tMaximum += !TE.slotNull(0) && TE.slotHas(0) && ST.item(TE.slot(0)) instanceof IItemReactorRod ? ((IItemReactorRod) ST.item(TE.slot(0))).getReactorRodNeutronMaximum(TE, 0, TE.slot(0)) : 0;
```
Comment (GregoriusT) [inline src/main/java/gregtech/tileentity/sensors/MultiTileEntityGeigerCounter.java:60]: Why did you use slotNull there? slotHas already does the null check. Did you look what slotNull does even?

File: src/main/java/gregtech/tileentity/energy/reactors/MultiTileEntityReactorRodBase.java [hunk 3]
```diff
@@ -45,34 +45,62 @@
...
-		new Textures.BlockIcons.CustomIcon("machines/generators/reactor_rods/colored/sides")
-	}, sOverlays[] = new IIconContainer[] {
-		new Textures.BlockIcons.CustomIcon("machines/generators/reactor_rods/overlay/top"),
-		new Textures.BlockIcons.CustomIcon("machines/generators/reactor_rods/overlay/sides")
-	};
-	
-	@Override public boolean setBlockBounds2(Block aBlock, int aRenderPass, boolean[] aShouldSideBeRendered) {box(aBlock, PX_P[ 6], PX_P[ 0], PX_P[ 6], PX_N[ 6], PX_N[ 0], PX_N[ 6]); return T;}
-	@Override public AxisAlignedBB getCollisionBoundingBoxFromPool() {return box(PX_P[ 6], PX_P[ 0], PX_P[ 6], PX_N[ 6], PX_N[ 0], PX_N[ 6]);}
-	@Override public AxisAlignedBB getSelectedBoundingBoxFromPool () {return box(PX_P[ 6], PX_P[ 0], PX_P[ 6], PX_N[ 6], PX_N[ 0], PX_N[ 6]);}
-	@Override public void setBlockBoundsBasedOnState(Block aBlock)  {box(aBlock, PX_P[ 6], PX_P[ 0], PX_P[ 6], PX_N[ 6], PX_N[ 0], PX_N[ 6]);}
```
Comment (GregoriusT) [inline src/main/java/gregtech/tileentity/energy/reactors/MultiTileEntityReactorRodBase.java:70]: There is still formatting changes you did not revert yet.

File: src/main/java/gregtech/tileentity/energy/reactors/MultiTileEntityReactorCore2x2.java [hunk 4]
```diff
@@ -52,131 +53,54 @@
  * @author Gregorius Techneticies
  */
 public class MultiTileEntityReactorCore2x2 extends MultiTileEntityReactorCore implements ITileEntitySwitchableMode {
```
Comment (GregoriusT) [inline src/main/java/gregtech/tileentity/energy/reactors/MultiTileEntityReactorCore2x2.java:55]: So many formatting changes that make the commit an absolute pain to read

File: src/main/java/gregtech/tileentity/energy/reactors/MultiTileEntityReactorCore1x1.java [hunk 5]
```diff
@@ -51,118 +52,56 @@
  * @author Gregorius Techneticies
  */
 public class MultiTileEntityReactorCore1x1 extends MultiTileEntityReactorCore {
```
Comment (GregoriusT) [inline src/main/java/gregtech/tileentity/energy/reactors/MultiTileEntityReactorCore1x1.java:54]: More random formatting changes that make it hard to see what changed.

File: src/main/java/gregapi/block/multitileentity/MultiTileEntityItemInternal.java (hunk 1 above)
Comment (Erik3003) [reply src/main/java/gregapi/block/multitileentity/MultiTileEntityItemInternal.java]: Shouldn't be breaking anything but I reverted it regardless

File: src/main/java/gregtech/tileentity/sensors/MultiTileEntityGeigerCounter.java (hunk 2 above)
Comment (Erik3003) [reply src/main/java/gregtech/tileentity/sensors/MultiTileEntityGeigerCounter.java]: Was a leftover from a failed fix, is now removed

File: src/main/java/gregtech/tileentity/energy/reactors/MultiTileEntityReactorRodBase.java (hunk 3 above)
Comment (Erik3003) [reply src/main/java/gregtech/tileentity/energy/reactors/MultiTileEntityReactorRodBase.java]: Formatting should be finally fixed in newest commit

**Comment by OvermindDL1** [comment]
This is looking quite nice.  ^.^

**Pull request merged**